addons/binding/org.openhab.binding.blebox/src/main/java/org/openhab/binding/blebox/internal/discovery/BleboxScanner.java

```diff
@@ -0,0 +1,239 @@
+
+    public BleboxScanner(BleboxDiscovery bleboxDiscovery) {
+        this.bleboxDiscovery = bleboxDiscovery;
+    }
+
+    /**
+     * Method for Bridge Discovery.
+     */
+    public synchronized void discoverBridge() {
+        logger.debug("Starting Blebox Discovery.");
+
+        SubnetUtils subnetUtils = null;
+        SubnetInfo subnetInfo = null;
+        long lowIP = 0;
+        long highIP = 0;
+
+        try {
+            List<Inet4Address> inet4 = getInet4Addresses();
+            logger.error("discoverBridge(): ip addresses - {}", inet4.size());
+            logger.error("discoverBridge(): ip main ip - {}", inet4.get(0));
+
+            NetworkInterface networkInterfacee = NetworkInterface.getByInetAddress(inet4.get(0));
+            logger.error("discoverBridge(): networkInterface - {}", networkInterfacee.toString());
+            String hostAddress = inet4.get(0).getHostAddress();
+            subnetUtils = new SubnetUtils(hostAddress + "/" + "24");
+            subnetInfo = subnetUtils.getInfo();
+            lowIP = convertIPToNumber(subnetInfo.getLowAddress());
+            highIP = convertIPToNumber(subnetInfo.getHighAddress());
+        } catch (IllegalArgumentException e) {
+            logger.error("discoverBridge(): Illegal Argument Exception - {}", e.toString());
+            return;
+        } catch (Exception e) {
+            logger.error("discoverBridge(): Error - Unable to get Subnet Information! {}", e.toString());
+            return;
+        }
+
+        logger.debug("   Local IP Address: {} - {}", subnetInfo.getAddress(),
+                convertIPToNumber(subnetInfo.getAddress()));
+        logger.debug("   Subnet:           {} - {}", subnetInfo.getNetworkAddress(),
+                convertIPToNumber(subnetInfo.getNetworkAddress()));
+        logger.debug("   Network Prefix:   {}", subnetInfo.getCidrSignature().split("/")[1]);
+        logger.debug("   Network Mask:     {}", subnetInfo.getNetmask());
+        logger.debug("   Low IP:           {}", convertNumberToIP(lowIP));
+        logger.debug("   High IP:          {}", convertNumberToIP(highIP));
+
+        ExecutorService threadpool = Executors.newFixedThreadPool(50);
+        Async async = Async.newInstance().use(threadpool);
+
+        for (long ip = lowIP; ip <= highIP; ip++) {
+            try {
+                final String ipAddress = convertNumberToIP(ip);
+
+                Gson gson = new Gson();
+
+                URIBuilder builder = new URIBuilder();
+                builder.setScheme("http").setHost(ipAddress).setPath("/api/device/state");
+                URI requestURL = null;
+                try {
+                    requestURL = builder.build();
+                } catch (URISyntaxException use) {
+                }
+
+                final Request request = Request.Get(requestURL).connectTimeout(2000).socketTimeout(2000);
+
+                Future<Content> future = async.execute(request, new FutureCallback<Content>() {
```

Comment:
Why call concurrently?

Reply:
I have to scan whole local subnet, in most cases we will hit timeout, so in single thread scanning it will take over 8 minutes. With 50 threads it will take 10 seconds.